midend/flattenLogMsg.cpp

```diff
@@ -0,0 +1,198 @@
+#include "midend/flattenLogMsg.h"
+#include "flattenLogMsg.h"
+
+namespace P4 {
+
+const IR::Type_StructLike* FindTypeInLogMsgForReplace::hasStructInParameter(
+        const IR::MethodCallStatement* methodCallStatement) {
+    if (auto* path = methodCallStatement->methodCall->method->to<IR::PathExpression>()) {
+        if (path->path->name.name == "log_msg") {
+            auto* param1 = methodCallStatement->methodCall->arguments->at(0)->expression;
```

Comment:
you should first check that there exist 2 arguments. Users are free to declare a log_msg function with a different signature. In that case you should give an error; your approach will crash the compiler if the signature does not have at least 2 arguments.